File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] first = {1, 2, 3, 4, 5};
        int[] second = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] third =  {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] count = {0, 0, 0};
        
        for (int i = 0; i < answers.length; i++) {
            if (answers[i] == first[i % 5]) {
                count[0]++;
            }
            if (answers[i] == second[i % 8]) {
                count[1]++;
            }
            if (answers[i] == third[i % 10]) {
                count[2]++;
            }
        }
        
        ArrayList<Integer> answer = new ArrayList<>();
        
        int max = Math.max(count[2], Math.max(count[0], count[1]));
        for (int i = 0; i < 3; i++) {
            if (count[i] == max) {
                answer.add(i+1);
            }
        }
                
        return answer.stream()
            .mapToInt(x->x)
            .toArray();
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    # 찍는 방식
    way1 = [1, 2, 3, 4, 5]
    way2 = [2, 1, 2, 3, 2, 4, 2, 5]
    way3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    # 정답을 모두 탐색하면서 찍은 번호와 맞는지 확인
    correct = [0, 0, 0]
    for i in range(len(answers)):
        if way1[i % 5] == answers[i]:
            correct[0] += 1
        if way2[i % 8] == answers[i]:
            correct[1] += 1
        if way3[i % 10] == answers[i]:
            correct[2] += 1
            
    return [index + 1 for index, value in enumerate(correct) if value == max(correct)]